plugins/org.eclipse.elk.alg.force/src/org/eclipse/elk/alg/force/model/AbstractForceModel.java

```diff
@@ -181,8 +180,26 @@ public static void avoidSamePosition(final Random random, final FParticle u, fin
         KVector pu = u.getPosition();
         KVector pv = v.getPosition();
         while (pu.x - pv.x == 0 && pu.y - pv.y == 0) {
-            pu.wiggle(random, 1);
-            pv.wiggle(random, 1);
+            boolean triedForBendPoints = false;
+            if (u instanceof FBendpoint && v instanceof FBendpoint && !triedForBendPoints) {
+                // Wiggle orthogonal to edge direction
+                FEdge uE = ((FBendpoint) u).getEdge();
+                KVector uV = new KVector(uE.getTargetPoint()).sub(uE.getSourcePoint());
```

Comment:
consider renaming uV to something to something more clear. uE seems self-explanatory, but uV not so much